### PR TITLE
Tests to check examples in the article

### PR DESCRIPTION
**Pull request by stitam**
PR task list:
- [x] Update NEWS
- [x] Add tests (if appropriate)
- [ ] (Update documentation with `devtools::document()`)
- [x] Check package passed

### CODE REVIEW

**Comment by stitam** [comment]
Also this PR closes #204.

**Comment by Aariq** [comment]
Aren't many of these tests duplicates of tests we already have?

**Comment by stitam** [comment]
Thanks @Aariq, are you refering to the examples in `test-etox.R`? I think those examples are from an earlier version and should be removed. It's probably better to put all tests in a separate test file.

**Comment by Aariq** [comment]
I just meant there's double coverage of functions if we add these tests. E.g. the test on line 123 just checks if `aw_query()` is working as expected—we already have a test for that in test-alanwood.R.  That's fine, but it's going to take the tests longer to run and I'm not sure I fully understand the purpose or necessity of these tests. Is it just so that people who come here from the publication find everything working exactly as expected?

**Comment by stitam** [comment]
I see. My reasoning was that we have to be more careful with tests that check outcomes in the article and only change them if absolutely necessary. For this we have to know that some tests are more imporant than the others, and I figured it's best to put them into another file and disregard duplicates. But I see your point regarding testing speed. I will try to add these tests to their original test files and remove duplicates.

**Comment by stitam** [comment]
I restructured the tests and tried to remove duplicates. etox and chemspider hasn't been tested yet as the servers seem to be down at the moment.

**Comment by stitam** [comment]
Some functions returned a warning about closing unused connections. I think this occurred when a scraper was asked to access a page that didn't exist (e.g. intentionally invalid query). I added `on.exit(suppressWarnings(closeAllConnections()))` to scrapers to ensure connections are closed without warning.

**Comment by stitam** [comment]
Also some parts of the article had to be edited, mostly because of multiple updates to `get_cid()`.

**Comment by Aariq** [comment]
> Also some parts of the article had to be edited, mostly because of multiple updates to `get_cid()`.

I promise not to drastically change outputs of functions for a little while at least...  Perhaps we should focus on bug-fixing (e.g. `ci_query`) and then make a release?

File: R/flavornet.R [hunk 1]
```diff
@@ -27,6 +27,7 @@
 fn_percept <- function(CAS, verbose = TRUE, ...)
 {
   foo <- function (CAS, verbose){
+    on.exit(suppressWarnings(closeAllConnections()))
```
Comment (Aariq) [inline R/flavornet.R:30]: Thank you so much for this.  I couldn't figure out how to deal with the weird straggler open connection warnings!

File: tests/testthat/test-utils.R [hunk 2]
```diff
@@ -1,12 +1,22 @@
 library(rcdk)
+
+test_that("examples in the article are unchanged", {
+  expect_false(is.inchikey("BQJCRHHNABKAKU-KBQPJGBKS-AN"))
+  expect_equal(capture_messages(is.inchikey("BQJCRHHNABKAKU-KBQPJGBKS-AN")),
+               "Hyphens not at position 15 and 26.\n")
+  expect_false(is.cas('64-17-6'))
+  expect_equal(
+    capture_messages(is.cas("64-17-6")), "Checksum is not correct! 5 vs. 6\n")
+  expect_false(is.inchikey("BQJCRHHNABKAKU-KBQPJGBKSA-5", type = "chemspider"))
```
Comment (Aariq) [inline tests/testthat/test-utils.R:10]: Does this need a `skip_if_not()` to check if chemspider is down?  I'm not super familiar with how this one works.

**Comment by stitam** [comment]
> I promise not to drastically change outputs of functions for a little while at least...

I am glad the `get_*()` functions are now consistent, it was definitely worth all the trouble.

File: tests/testthat/test-utils.R (hunk 2 above)
Comment (stitam) [reply tests/testthat/test-utils.R:10]: Thanks, I didn't notice this function accesses a web service.

**Comment by stitam** [comment]
Some tests seem to fail randomly on AppVeyor or Travis. All of these tests work locally..

**Comment by Aariq** [comment]
The appveyor failure seems like maybe a Windows + curl issue that's not unique to R: https://github.com/certbot/certbot/issues/7322.  The Travis one looks like it's sometimes just not connecting to CTS?  CTS has been quite slow lately.  Maybe the timeout needs to be longer for this function?
